<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>AuthKeysConfigurationPage</class>
 <widget class="QWidget" name="AuthKeysConfigurationPage">
  <property name="windowTitle">
   <string>Authentication keys</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../../core/core.qrc">
    <normaloff>:/resources/application-x-pem-key.png</normaloff>:/resources/application-x-pem-key.png</iconset>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QGroupBox" name="groupBox_5">
     <property name="title">
      <string>Introduction</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <property name="leftMargin">
       <number>10</number>
      </property>
      <item>
       <widget class="QLabel" name="label">
        <property name="text">
         <string>An authentication key pair consist of two coupled cryptographic keys, a private and a public key. A private key allows users on the master computer to access client computers. It is important that only authorized users have read access to the private key file. The public key is used on client computers to authenticate incoming connection request.

Technically the client sends a challenge (random data) for which the master has to create a cryptographic signature using the private key. The signature is sent back to the client which can verify the validity of the signature using the corresponding public key.</string>
        </property>
        <property name="textFormat">
         <enum>Qt::PlainText</enum>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_7">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="keyManagement">
     <property name="title">
      <string>Key file directories</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_12">
      <property name="verticalSpacing">
       <number>2</number>
      </property>
      <item row="0" column="0" colspan="2">
       <layout class="QGridLayout" name="gridLayout_8">
        <property name="horizontalSpacing">
         <number>10</number>
        </property>
        <property name="verticalSpacing">
         <number>7</number>
        </property>
        <item row="0" column="0">
         <widget class="QLabel" name="label_7">
          <property name="text">
           <string>Public key file base directory</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLineEdit" name="publicKeyBaseDir"/>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string>Private key file base directory</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLineEdit" name="privateKeyBaseDir"/>
        </item>
        <item row="0" column="2">
         <widget class="QToolButton" name="openPublicKeyBaseDir">
          <property name="text">
           <string>...</string>
          </property>
          <property name="icon">
           <iconset resource="../../core/core.qrc">
            <normaloff>:/resources/document-open.png</normaloff>:/resources/document-open.png</iconset>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QToolButton" name="openPrivateKeyBaseDir">
          <property name="text">
           <string>...</string>
          </property>
          <property name="icon">
           <iconset resource="../../core/core.qrc">
            <normaloff>:/resources/document-open.png</normaloff>:/resources/document-open.png</iconset>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Available authentication keys</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QTableView" name="keyTable">
        <property name="editTriggers">
         <set>QAbstractItemView::NoEditTriggers</set>
        </property>
        <property name="selectionMode">
         <enum>QAbstractItemView::SingleSelection</enum>
        </property>
        <property name="selectionBehavior">
         <enum>QAbstractItemView::SelectRows</enum>
        </property>
        <attribute name="horizontalHeaderStretchLastSection">
         <bool>true</bool>
        </attribute>
        <attribute name="verticalHeaderVisible">
         <bool>false</bool>
        </attribute>
       </widget>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QPushButton" name="createKeyPair">
          <property name="text">
           <string>Create key pair</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="deleteKey">
          <property name="text">
           <string>Delete key</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="importKey">
          <property name="text">
           <string>Import key</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="exportKey">
          <property name="text">
           <string>Export key</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="setAccessGroup">
          <property name="text">
           <string>Set access group</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer_2">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="../../core/core.qrc"/>
  <include location="../../core/core.qrc"/>
  <include location="../../core/core.qrc"/>
 </resources>
 <connections/>
</ui>
